src/app/InteractionModelEngine.cpp

```diff
@@ -329,6 +379,19 @@ CHIP_ERROR InteractionModelEngine::SendReadRequest(ReadPrepareParams & aReadPrep
     return err;
 }
 
+CHIP_ERROR InteractionModelEngine::SendSubscribeRequest(ReadPrepareParams & aReadPrepareParams, uint64_t aAppIdentifier)
+{
+    ReadClient * client = nullptr;
+    CHIP_ERROR err      = CHIP_NO_ERROR;
+    ReturnErrorOnFailure(NewReadClient(&client, ReadClient::InteractionType::Subscribe, aAppIdentifier));
+    err = client->SendSubscribeRequest(aReadPrepareParams);
```

Comment:
similar here: a pattern of 'on error the caller is responsible for shutdown' seems not as convenient as 'call the method and be done'.

I can see spearating shutdown lifetime from other operations as providing more control, but wondering if it is useful. A blanket 'on any error, client becomes invalid' seems to give some consistency and maybe simplify APIs.

This can be done in future PRs though. I would prefer not to completely derail something that works.

Reply:
move shutdown inside SendSubscribeRequest, thanks